### DISCUSSION TITLE
Robotics Academy Challlenge Confusion

### DISCUSSION THREAD

**Discussion by lavish123409** [JdeRobot/jderobot.github.io#268] [Q&A]
Hello everyone,

I am trying to do the Robotics Academy challenge from ideas list. It is saying to install Robotics Academy on my machine and solve several of its exercises **(ONLY of release 2.3, the one with web based templates).** I am confused whether I should clone the repo and do the exercises on browser or download [this](https://github.com/JdeRobot/RoboticsAcademy/releases/tag/v2.3.2) release and write solution in `MyAlgorithm.py`.

If someone can clear this confusion, then it would be very helpful.
Thanks

**Comment by pawanw17** [comment]
Hi @lavish123409 the challenge requires you to build/run Robotics Academy on Docker and solve an exercise

**Reply by pawanw17** [reply]
see here: https://youtu.be/URHuhA7FIfg?si=wt6eHYmhHZ8DoYQO

**Reply by lavish123409** [reply]
Hey @pawanw17, when I am trying to run docker with the GPUs, using
```bash
docker run --rm -it --gpus all --device /dev/dri -p 7164:7164 -p 2303:2303 -p 1905:1905 -p 8765:8765 -p 6080:6080 -p 1108:1108 -p 7163:7163 jderobot/robotics-academy
```

It is giving following error.
`docker: Error response from daemon: error gathering device information while adding custom device "/dev/dri": no such file or directory.` You have any idea how to run the docker using GPUs, because it is not running steadily in my machine.

Thanks a lot for your help btw.

**Reply by lavish123409** [reply]
Hey @pawanw17, I resolved that error, now it is running fine on my machine.

But, I am not able to use the web interface properly. I have coded a solution but it is not giving correct output. For debugging purposes, I am trying to print some values while running the code, but because of while loop it is printing continuously and on clicking the pause button the simulation is getting paused not the code(might be a bug or I would not be knowing how to use it). 

Is there any way that I can run the code directly on docker container along with the simulation where I can debug easily?

Thanks

**Reply by lavish123409** [reply]
Any suggestions @sergiopaniego

**Reply by pawanw17** [reply]
@lavish123409, I don't think there's a direct way to run the code outside the web interface, can you share me more details of what issue you are facing while running the code?

**Reply by lavish123409** [reply]
Yes @pawanw17, when I am trying to print the output in the terminal for debugging, it is continuously printing the values because of the iterative code. So, because of that I am not able to see where my code is going wrong. I thought by the running my code on terminal, I can stop the code anytime or maybe print the needed values.
